test/controllers/rentals_controller_test.rb

```diff
@@ -0,0 +1,60 @@
+require "test_helper"
+
+describe RentalsController do
+  describe "check_out" do
+    it "creates a new instance of Rental if the movie is available" do
+      movie = movies(:movie1)
+      customer = customers(:customer1)
+
+      new_rental = {
+        customer_id: customer.id,
+        movie_id: movie.id,
+      }
+      expect {
+        post check_out_path(new_rental)
+      }.must_differ "Rental.count", 1
+
+      expect(new_rental[:customer_id]).must_equal customer.id
+      expect(new_rental[:movie_id]).must_equal movie.id
+      body = JSON.parse(response.body)
+      expect(body).must_be_instance_of Hash
+    end
+
+
+    it "returns an error of Bad Request given bad rental data" do
+      new_rental = nil
+
+      expect { 
+        post check_out_path(new_rental)
+      }.wont_change "Rental.count"
+
+      must_respond_with :bad_request
+      body = JSON.parse(response.body)
+      expect(body['errors']).must_include "Bad Request"
+    end
+  end
+
+  describe "check_in" do
```

Comment:
What about invalid check ins and checkouts?  Trying to check out a movie with no more inventory, or checking in a movie that's not checked out.  More testing needed.